backend/dataall/cdkproxy/assets/datalakelocationcustomresource/index.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import os
+import json
+import boto3
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get("LOG_LEVEL", "INFO"))
+log = logging.getLogger(__name__)
+
+lf_client = boto3.client("lakeformation", region_name=os.environ.get("AWS_REGION"))
+lambda_client = boto3.client("lambda", region_name=os.environ.get("AWS_REGION"))
+
+
+def on_event(event, context):
+    request_type = event["RequestType"]
+    if request_type == "Create":
+        return on_create(event)
+    if request_type == "Update":
+        return on_update(event)
+    if request_type == "Delete":
+        return on_delete(event)
+    raise Exception(f"Invalid request type: {request_type}")
+
+
+def on_create(event):
+    """ Checks if the S3 location is already registered in Lake Formation.
+    If already registered it updated the roleArn for the location.
+    If not registered, it registers the location.
+    """
+    props = event["ResourceProperties"]
+    lambda_policy_remove_duplicates(props)
```

Comment:
Why do we have duplicates? Is there a way to avoid creation of duplicated policy for lambda not to delete this duplication after?

Reply:
Looks like we are doing it to replace the old manual pivot role. Is that correct?

Reply:
Nope, check the conversation above